src/reactive/observable.ts

```diff
@@ -0,0 +1,102 @@
+// notice_start
+/*
+ * Copyright 2015 Dev Shop Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// notice_end
+
+import {Guard} from '../system';
+import {Observer} from './observer';
+import {Subscribe} from './subscribeDelegate';
+import {RouterObservable} from './routerObservable';
+import {Router} from '../router';
+import {DisposableWrapper} from '../system/disposables';
+import {DictionaryDisposable} from '../system/disposables';
+import {DisposableOrFunction} from '../system/disposables/disposable';
+import {AutoConnectedObservable} from './autoConnectedObservable';
+
+export interface Observable<T> {
+    asObservable?(): Observable<T>;
+    do?(action: (item: T) => void): Observable<T>;
+    map?(action: (item: T) => any): Observable<T>;
```

Comment:
I think this is incorrect.  should be:
`map<TResult>(action: (item:T) => TResult): Observable<TResult>` otherwise the mapping has to return the same type.